Source/ACE.DatLoader/DatReader.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.IO;
+
+namespace ACE.DatLoader
+{
+    public class DatReader
+    {
+        public int Offset { get; set; }
+        private byte[] buffer;
+
+        public DatReader(string datFilePath, uint offset, uint size, uint sectorSize)
+        {
+            uint nextAddress = 0;
+            FileStream stream = new FileStream(datFilePath, FileMode.Open, FileAccess.Read);
+            using (stream)
+            {
+                this.buffer = new byte[size];
+                stream.Seek(offset, SeekOrigin.Begin);
+
+                if (size > sectorSize)
+                    nextAddress = this.GetNextAddress(stream, -4);
+
+                int bufferOffset = 0;
+                while (size > 0)
+                {
+                    if (size < sectorSize)
+                    {
+                        stream.Read(buffer, bufferOffset, Convert.ToInt32(size));
+                        size = 0;
+                    }
+                    else
+                    {

```

Comment:
I'd really like to see this block commented fairly heavily.  It took me several minutes of staring at it to grok what it is doing, where comments would have taken 20 seconds to read to get the explanation.